### ISSUE TITLE
Personalized Augmentation Method

### ISSUE THREAD

**Issue by javierpastorfernandez** [stardist/stardist#137]
Good evening,

I have developed my own Augmentation method, trying to implement more aggressive transformations of the input image, in a attempt to achieve better generalization results.

I have used the library imgaug (Source: https://imgaug.readthedocs.io/en/latest/index.html)
Nevertheless, many of the transformations do need the images to be in uint8 data type [0,255], which is incompatible with the input of the UNet (float)([0,1]). I have tried to solve this by getting rid of the normalization step, and introducing it in the "augmenter " function.
 
![image](https://user-images.githubusercontent.com/60293572/116792791-36394680-aac3-11eb-9e5c-c35f7db3a0d5.png)

I thought that would work . However, there might be some internal processes of StarDist that clip/normalize the image before calling "augmenter"; which gives the error associated with imgaug.
![image](https://user-images.githubusercontent.com/60293572/116792832-67197b80-aac3-11eb-9d2a-d43924abafc5.png)

I would really appreciate it if you could help me a little bit ! 

Thank you in advance!! This is a great software :)

**Comment by maweigert** [comment]
Hi @javierpastorfernandez ,

Yes, stardist will convert the input images to `float32` before applying the augmenter, but does not normalize them. So simply converting them back to `int` at the beginning of your augmenter function should work. 

Hope that helps,
M

PS: We also recently made public a [small augmentation library](https://github.com/stardist/augmend) that we typically use with stardist and which might be of interest.

**Comment by javierpastorfernandez** [comment]
Hi,
First of all. Thank you really much for your response and time!

After casting the images to "int", I could at least start training.
Nevertheless, the training process seem to fail at some point as for some epochs the loss function goes to 10^18 and it recovers the normal values for some other epochs.

The augmentation function transforms the images and then uses the "normalizing" function and "fill_holes" from stardist.
1. If I get rid of the "normalizing function" inside my augmenter function,  the tranining losses are the usual ones.
2.  If I first use the augmenter function to transform the dataset, and then train without the any augmentation, the resulting training works correctly. 

Error: 
![image](https://user-images.githubusercontent.com/60293572/116812603-cfae3a00-ab4f-11eb-9178-8b4ec6d41fc9.png)

Case 2: First augment the dataset; Do not use it for training: 
![image](https://user-images.githubusercontent.com/60293572/116812617-e2287380-ab4f-11eb-9a7b-f7936bdea584.png)
![image](https://user-images.githubusercontent.com/60293572/116812619-e5bbfa80-ab4f-11eb-9423-c660723766ee.png)

After trying several things and inspecting the resulting images that feed the training, I thing it could be related with the way stardist handles the "augmenter" function? I am not sure. 

Thank you in advance :)

**Comment by javierpastorfernandez** [comment]
Case 2 images : (Previous images does not have good quality)

![image](https://user-images.githubusercontent.com/60293572/116812766-c40f4300-ab50-11eb-83c0-7be22455633f.png)
![image](https://user-images.githubusercontent.com/60293572/116812772-c83b6080-ab50-11eb-904a-310b1963c92b.png)

**Comment by uschmidt83** [comment]
Please share your code/notebook with us that we can take a look ourselves and ideally run it ourselves.

**Comment by javierpastorfernandez** [comment]
I am sending the augmenter function itself, as well as the notebook I am currently using for training :)

[Github.zip](https://github.com/stardist/stardist/files/6422618/Github.zip)

**Comment by uschmidt83** [comment]
Sorry, this is too difficult/time-consuming to debug, especially since we can't run the code ourselves.

**Comment by javierpastorfernandez** [comment]
Okay! Thank you anyways for your time! I really appreciate it
